infra/Pos/Network/Windows/DnsDomains.hs

```diff
@@ -0,0 +1,23 @@
+{-# LANGUAGE CPP                      #-}
+{-# LANGUAGE ForeignFunctionInterface #-}
+
+#if defined(mingw32_HOST_OS)
```

Comment:
I think all the CPP in this module can be avoided; just don't include the module at all when not on WIndows. Then we'd have to use CPP to import it where we need it, but in that part of the code we'll need CPP anyway.